src/solc_0.8/raffle/PeopleOfCrypto.sol

```diff
@@ -0,0 +1,267 @@
+        return
+            ECDSA.recover(
+                keccak256(
+                    abi.encodePacked(
+                        "\x19Ethereum Signed Message:\n32",
+                        keccak256(abi.encode(_wallet, _signatureId, _contractAddress, _chainId))
+                    )
+                ),
+                _signature
+            );
+    }
+
+    function checkPersonalizationSignature(
+        address _wallet,
+        uint256 _signatureId,
+        address _contractAddress,
+        uint256 _chainId,
+        uint256 _tokenId,
+        uint256 _personalizationMask,
+        bytes memory _signature
+    ) public pure returns (address) {
+        return
+            ECDSA.recover(
+                keccak256(
+                    abi.encodePacked(
+                        "\x19Ethereum Signed Message:\n32",
+                        keccak256(
+                            abi.encode(
+                                _wallet,
+                                _signatureId,
+                                _contractAddress,
+                                _chainId,
+                                _tokenId,
+                                _personalizationMask
+                            )
+                        )
+                    )
+                ),
+                _signature
+            );
+    }
+
+    // Thx Cyberkongs VX <3
+    function getRandomToken(address _wallet, uint256 _totalMinted) private returns (uint256) {
+        uint256 remaining = MAX_SUPPLY - _totalMinted;
+        uint256 rand =
+            uint256(keccak256(abi.encodePacked(_wallet, block.difficulty, block.timestamp, remaining))) % remaining;
+        uint256 value = rand;
+
+        if (availableIds[rand] != 0) {
+            value = availableIds[rand];
+        }
+
+        if (availableIds[remaining - 1] == 0) {
+            availableIds[rand] = remaining - 1;
+        } else {
+            availableIds[rand] = availableIds[remaining - 1];
+        }
+
+        return value;
+    }
+
+    function toggleSale() external onlyOwner {
+        paused = paused == 0 ? 1 : 0;
+        emit TogglePaused(paused == 1);
+    }
+
+    function personalize(
+        uint256 _signatureId,
+        bytes memory _signature,
+        uint256 _tokenId,
+        uint256 _personalizationMask
+    ) external {
+        require(ownerOf(_tokenId) != _msgSender(), "You must be the owner of the token in order to personalize it");
+
+        require(personalizationSignatureIds[_signatureId] == 0, "SignatureId already used");
+        require(
+            checkPersonalizationSignature(
+                _msgSender(),
+                _signatureId,
+                address(this),
+                block.chainid,
+                _tokenId,
+                _personalizationMask,
+                _signature
+            ) == signAddress,
+            "Signature failed"
+        );
+
+        personalizationSignatureIds[_signatureId] = 1;
```

Comment:
@ioanaverebi wondering if we really need personalizationSignatureIds, maybe we can reuse the same map from mint here:
signatureIds[_signatureId] = 1;

the idea would be to have same _signatureId generator from the raffle middleware for this to work (for both mint/personalize)

Reply:
Fair enough, it's a different signatureId for personalization anyways.